SU-HSDO/suhumsci
# feat (STN- 645): Convert imported D7 classes

# READY FOR REVIEW

## Summary
This a new feature to convert CSS classes from our D7 platform and legacy theme using the theme for imported content. 

**Scenario:**
- When importing content from D7 sites or switching themes from the legacy theme, we would like styles to be converted. 
- This is necessary because we import 100s+ nodes of content that manual conversion is not realistic

Class list below:

D7 class | D8 MRC legacy class | D8 class | D8 MRC legacy class
-- | -- | -- | --
caption | decanter-caption | hs-caption | decanter-caption
credits | decanter-credit | hs-credits | decanter-credit
descriptor |   | hb-descriptor |  
highlight-block-left |   | hb-blockquote |  
highlight-block-right |   | hb-blockquote |  
infotext | decanter-font-splash | hs-font-splash | decanter-font-splash
pullquote |   | hs-well |  
summary | decanter-font-lead | hs-font-lead | decanter-font-lead
btn | decanter-button | hs-button | decanter-button
~~more-link~~ | decanter-more-link | hs-more-link | decanter-more-link
private-link |   | hs-private-link |  

_**Note**: `more-link` has been removed from this work as it has already been restyled to work with previous work._


## Need Review By (Date)
asap

## Urgency
medium

## Steps to Test
1. npm run test
2. To confirm the classes all react the same you'll want to create a flexibile page and then add content to it using the classes.
- Here is a code block you could use for testing:


```
<h2>Captions:</h2>
<p class="hs-caption">hs-caption</p>
<p class="decanter-caption">decanter-caption</p>
<p class="caption">caption</p>
<hr />
<h2>Credits:</h2>
<p class="hs-credits">hs-credits</p>
<p class="decanter-credit">decanter-credit</p>
<p class="credits">credits</p>
<hr />
<h2>Descriptor:</h2>
<p class="hb-descriptor">hb-descriptor</p>
<p class="descriptor">descriptor</p>
<hr />
<h2>Highlight Block Left:</h2>
<p class="hb-blockquote">hb-blockquote on the p tag</p>
<blockquote>
<p>p tag within hb-blockquote</p>
</blockquote>
<p class="highlight-block-left">highlight-block-left on the p</p>
<div class="highlight-block-left"><p>highlight-block-left on a div with a p tag within</p></div>
<hr />
<h2>Highlight Block Right:</h2>
<p class="hb-blockquote">hb-blockquote on the p tag</p>
<blockquote>
<p>p tag within a blockquote</p>
</blockquote>
<p class="highlight-block-right">highlight-block-right on the p tag</p>
<div class="highlight-block-right"><p>highlight-block-right on a div with a p tag within</p></div>
<hr />
<h2>Info Text:</h2>
<p class="hs-font-splash">hs-font-splash</p>
<p class="decanter-font-splash">decanter-font-splash</p>
<p class="infotext">infotext</p>
<hr />
<h2>Well/Pullquote:</h2>
<p class="hs-well">hs-well</p>
<p class="pullquote">pullquote</p>
<hr />
<h2>Summary/Font lead:</h2>
<p class="hs-font-lead">hs-font-lead</p>
<p class="decanter-font-lead">decanter-font-lead</p>
<p class="summary">summary</p>
<hr />
<h2>Button:</h2>
<p class="hs-button"><a href="http://google.com">hs-button on the p tag</a></p>
<p class="decanter-button"><a href="http://google.com">decanter-button on the p tag</a></p>
<p class="btn"><a href="http://google.com">btn on the p tag</a></p>
<p>--</p>
<p><a class="hs-button" href="http://google.com">hs-button on the a tag</a></p>
<p><a class="decanter-button" href="http://google.com">decanter-button on the a tag</a></p>
<p><a class="btn" href="http://google.com">btn on the button tag</a></p>
<p>--</p>
<button class="hs-button" href="http://google.com">hs-button on the button tag</button>
<button class="decanter-button" href="http://google.com">decanter-button on the button tag</button>
<button class="btn" href="http://google.com">btn on the a tag</button>
<p>--</p>
<p><button class="hs-button" href="http://google.com">hs-button on the button tag within a p tag</button></p>
<p><button class="decanter-button" href="http://google.com">decanter-button on the button tag within a p tag</button></p>
<p><button class="btn" href="http://google.com">btn on the button tag within a p tag</button></p>
<hr />
<h2>More Link:</h2>
<p class="hs-more-link"><a href="http://google.com">hs-more-link on the p tag</a></p>
<p class="decanter-more-link"><a href="http://google.com">decanter-more-link on the p tag</a></p>
<p>--</p>
<p><a class="hs-more-link" href="http://google.com">hs-more-link on the a tag</a></p>
<p><a class="decanter-more-link" href="http://google.com">decanter-more-link on the a tag</a></p>
<hr />
<h2>Private Links:</h2>
<p class="hs-private-link"><a href="http://google.com">hs-private-link on the p tag</a></p>
<p class="private-link"><a href="http://google.com">private-link on the p tag</a></p>
<p>--</p>
<p><a class="hs-private-link" href="http://google.com">hs-private-link on the a tag</a></p>
<p><a class="private-link" href="http://google.com">private-link on the a tag</a></p>
```



- There are heading texts and break lines between each area to help with organization and also for additions like the button tests, I've added multiple ways they can be added to the page. To add this without the WYSIWYG tampering with your code, you'll want to make sure your CKEditor is set to Full HTML, and swapping to Source and then saving it will keep the Drupal CKEditor from tampering with your testing elements.

3. You'll also want to test this using views, to test these classes open up any View and add the imported classes as a field class. You can do this here with the following path (Edit View > Click on the Views Field you want to edit > Open up the Style Settings drop-down > Check "Customize field HTML" > Check "Create a CSS class" > Add class to the text field). After adding that class, save the view and you should see the field update using that class you've added.
![image](https://user-images.githubusercontent.com/9355698/107286821-18fa5c00-6a2f-11eb-8313-d3272c729831.png)
- You can also test this a little bit differently putting a div wrapper around these elements with the class "views-row" on that wrapper element.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
